src/esp/gfx/GenericDrawable.cpp

```diff
@@ -38,31 +38,55 @@ void GenericDrawable::setLightSetup(const Mn::ResourceKey& resourceKey) {
   updateShader();
 }
 
-void GenericDrawable::draw(const Mn::Matrix4& transformationMatrix,
-                           Mn::SceneGraph::Camera3D& camera) {
-  updateShader();
-
+void GenericDrawable::updateShaderLightingParameters(
+    const Mn::Matrix4& transformationMatrix,
+    Mn::SceneGraph::Camera3D& camera) {
   const Mn::Matrix4 cameraMatrix = camera.cameraMatrix();
 
-  std::vector<Mn::Vector3> lightPositions;
+  // todo: move to MaterialHelper or elsewhere
+  std::vector<Mn::Vector4> lightPositions;
   lightPositions.reserve(lightSetup_->size());
-  std::vector<Mn::Color4> lightColors;
+  std::vector<Mn::Color3> lightColors;
   lightColors.reserve(lightSetup_->size());
+  std::vector<Mn::Color3> lightSpecularColors;
+  lightSpecularColors.reserve(lightSetup_->size());
+  constexpr float dummyRange = 10000;
+  std::vector<float> lightRanges(lightSetup_->size(), dummyRange);
```

Comment:
Coming soon, we're going to import lights from json. At that time, we can include a range per light instead of my hard-coded dummy value.